Event delete/add, join/leave

alivivi opened the pull request:
-Once an event clicked, it will direct to eventDescription page with all info
-Only the user created the event can delete it, the DELETE EVENT button will only be shown to event creator
-Once a user join an event, it will be added on participants textview for eventDescription page, and the JOIN EVENT button will be changed to LEAVE EVENT
-Once a user leave the event, the users name will be deleted from participants and the button will be changed to JOIN EVENT
-For the events on homepage, just show COURSE and SUBJECT
-For the user who create the event, use REVIEW EVENT to event description page, since the event creator should not JOIN/LEAVE EVENT
-For the events created by other user, with a ADD FRIEND button on event description page
-Once an event deleted, all participants in that event will also be deleted
-After clicked on course buttons, home page will filter the event for that course only

alivivi committed:
Event delete/add, join/leave

-Once an event clicked, it will direct to eventDescription page with all info
-Only the user created the event can delete it, the DELETE EVENT button will only be shown to event creator
-Once a user join an event, it will be added on participants textview for eventDescription page, and the JOIN EVENT button will be changed to LEAVE EVENT
-Once a user leave the event, the users name will be deleted from participants and the button will be changed to JOIN EVENT

alivivi committed:
modification on event scenario

-For the events on homepage, just show COURSE and SUBJECT
-For the user who create the event, use REVIEW EVENT to event description page, since the event creator should not JOIN/LEAVE EVENT
-For the events created by other user, with a ADD FRIEND button on event description page
-Once an event deleted, all participants in that event will also be deleted

Leksha commented on StudyBuddy/app/src/main/java/uw/studybuddy/Events/EventCardViewHolder.java:
```diff
@@ -14,25 +24,65 @@
+
+        BjoinEvent = (Button)mView.findViewById(R.id.BjoinEvent);
+
+        mDatabaseJoinEvent = FirebaseDatabase.getInstance().getReference().child("Participants");
```
Could you move this to the `FirebaseInstace` class? We just want to keep everything that is database related into that class.

Leksha reviewed (approved):
Tested the feature, no issues found.

alivivi committed:
Filter Event

-After clicked on course buttons, home page will filter the event for that course only

Leksha committed:
Merge branch 'master' into Event&Participants

Leksha committed:
Fixed bugs in EventsListRecycleViewFragment

Leksha committed:
Deleted firebase_library
